woocommerce/woocommerce
# Disable checkout menu when checkout block is default

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #44120.

This PR adds the following changes:

- Disable the checkout menu item in the Customizer if the merchant is using a class theme and the Checkout block is the default checkout.
- Add a notification to the Customizer with a link that points to the editor page that contains the Checkout block.

> [!NOTE]
> The acceptance criteria on #44120 mention _"If the store has a block theme, the user should be redirected to the template"_. As the Customizer is not visible when using a block theme, I've skipped this acceptence criteria.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate the [Storefront](https://wordpress.org/themes/storefront/) theme.
2. Go to `WooCommerce » Settings » Advanced` and set the page with the Checkout block as the checkout page.
3. Go to `Appearance » Customize » WooCommerce`.
4. Verify that the Checkout menu item is disabled.
5. Verify that the notice `Checkout can be customized in the Editor with your active theme.` is visible.
6. Verify that the link `in the Editor` points to the editor page that contains the Checkout block.
7. Go to `WooCommerce » Settings » Advanced` and set the page with the checkout shortcode as the checkout page.
8. Go to `Apperance » Customize » WooCommerce`.
9. Verify that the Checkout menu item is not disabled and can be accessed.
10. Verify that the notice `Checkout can be customized in the Editor with your active theme.` is not visible. 
11. Activate the [Twenty Twenty-Four](https://wordpress.org/themes/twentytwentyfour/) theme.
12. Verify that `Appearance » Customize` is not visible.

<table>
<tr>
<td valign="top">Before:
<br><br>
<img width="302" alt="Screenshot 2024-03-13 at 14 54 57" src="https://github.com/woocommerce/woocommerce/assets/3323310/7b297a86-0faf-4df7-9322-6f20e4bb6569">
</td>
<td valign="top">After:
<br><br>
<img width="302" alt="Screenshot 2024-03-13 at 14 54 28" src="https://github.com/woocommerce/woocommerce/assets/3323310/01e2d080-3702-4c8f-a967-2dc691ae0756">
</td>
</tr>
</table>



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Disable the checkout menu within Customizer when the checkout block is the default checkout page.

#### Comment 

</details>
